### ISSUE TITLE
Regression between 3dfb4ea and 7f4d25a

### ISSUE THREAD

**Issue by prognostikos** [kristianmandrup/cantango#11]
Unfortunately I'm seeing another NoMethodError with the combination of activeadmin and cantango at the latest 7f4d25a1d11d8f3012096ea310acfbc4588bfd7b.

3dfb4ea33db2ca8f6ec5b0c1f9fb8a7c633f0980 of cantango tested fine yesterday. With the latest versons of master of both active admin and cantango I'm seeing the following error if both are enabled. If either or neither are enabled, there are no problems.

```
lib/kaminari/models/mongoid_extension.rb:22:in `block in <module:Document>': undefined method `scope' for Mongoid::Document:Module (NoMethodError)
```

I realize the source of the error is kaminari which is required by active admin, but again using active admin alone I do not see the same error, and active admin + cantango 3dfb4ea33db2ca8f6ec5b0c1f9fb8a7c633f0980 works fine as well. It's only when I have both bundled with cantango at 7f4d25a1d11d8f3012096ea310acfbc4588bfd7b that I see the exception.

**Comment by prognostikos** [comment]
BTW I'm not blocked as I've added

```
gem 'cantango', :git => 'https://github.com/kristianmandrup/cantango.git', :ref => '3dfb4ea'
```

to my `Gemfile`

**Comment by kristianmandrup** [comment]
$ ack Mongoid

I think it has something todo with the ORM adapters I have recently added. Didn't really test them yet!
See if this gives you a clue to the problem.

<pre>
lib/cantango/adapter/init.rb
1:class MongoidModels
7:module Mongoid
10:      MongoidModels.all << klass
</pre>


<pre>
lib/cantango/configuration/models/mongoid.rb
4:      class Mongoid < Generic
6:          MongoidModels.all
</pre>


<pre>
lib/cantango/configuration/models.rb
4:      autoload_modules :Generic, :ActiveRecord, :DataMapper, :MongoMapper, :Mongoid
</pre>


<pre>
lib/cantango/rules/adaptor/mongo.rb
6:        # using #in on Hash (Mongoid query)
</pre>


<pre>
lib/cantango/rules/adaptor/mongoid.rb
4:      module Mongoid
</pre>


<pre>
lib/cantango/rules/adaptor.rb
5:      autoload_modules :ActiveRecord, :DataMapper, :Mongoid, :MongoMapper
30:          :mongoid => "Mongoid::Document",
</pre>


Thanks :)

**Comment by kristianmandrup** [comment]
Or perhaps:

<pre>
lib/cantango/rules/dsl.rb
19:      def scope name, &block
</pre>


This method should be renamed to not conflict with Rails built-in scope method!

**Comment by kristianmandrup** [comment]
I figured out the problem. It was due to an intrusive way I had hacked into self.included of Mongoid::Document in order to get a list of all the collections (tables) of the current Mongo DB. I've now found a simpler non-intrusive way to do this.
I'm currently refactoring CanTango engines with respect to caching to it more "intelligent" and fine-grained (optimized).

**Comment by kristianmandrup** [comment]
Hi @prognostikos, I have just released a version 0.9.4 which should fix the ORM adapter problem you encountered when integrating with Kaminari. The master branch has also been updated to reflect the latest changes and improvements.
I will start building a demo app now, and I plan to integrate Active Admin and the CanTango permissions editor in the process... did you get further along with your app?

Kris
